go/vt/vtgate/vtgate_test.go

```diff
@@ -2683,3 +2683,45 @@ func TestVTGateShowMetadataTwoShards(t *testing.T) {
 		t.Errorf("shard %s not found in Values \n%+v", shard1, qr.Rows)
 	}
 }
+
+func TestParseKeyspaceOptionalShard(t *testing.T) {
+	testcases := []struct {
+		keyspaceShard string
+		keyspace      string
+		shard         string
+	}{{
+		keyspaceShard: "ks",
+		keyspace:      "ks",
+		shard:         "",
+	}, {
+		keyspaceShard: "/-80",
```

Comment:
I suggest that we remove the test cases where one of the parts is empty.

IMO this is outside of the "specification" and therefore the behaviour for such an input should not be defined. But by having it in the test, we're implicitly defining it and making it part of our "specification". For example, we may consider to error on this input in the future.

Therefore, I prefer if you delete this case and the cases `ks/`, `ks:` and `:-80`.